fdbclient/NativeAPI.actor.cpp

```diff
@@ -3373,3 +3373,56 @@ ACTOR Future<Void> snapCreate(Database cx, Standalone<StringRef> snapCmd, UID sn
 		throw;
 	}
 }
+
+ACTOR Future<bool> checkSafeExclusions(Database cx, vector<AddressExclusion> exclusions) {
+	TraceEvent("ExclusionSafetyCheckBegin")
+	    .detail("NumExclusion", exclusions.size())
+	    .detail("Exclusions", describe(exclusions));
+	state ExclusionSafetyCheckRequest req(exclusions);
+	state bool ddCheck;
+	loop {
+		choose {
+			when(wait(cx->onMasterProxiesChanged())) {}
+			when(ExclusionSafetyCheckReply _ddCheck = wait(loadBalance(
+			         cx->getMasterProxies(false), &MasterProxyInterface::exclusionSafetyCheckReq, req, cx->taskID))) {
+				ddCheck = _ddCheck.safe;
```

Comment:
Since this request's reply can send error, we need to handle the error. Refer to `snapCreate` and `splitStorageMetrics` actor above.